src/error.rs

```diff
@@ -0,0 +1,37 @@
+use failure::Fail;
+
+#[derive(Debug, Fail)]
+pub enum Error {
```

Comment:
I'm somewhat surprised to see you choose a single error enum as dcc is a large and complex lib, but then I still struggle with rust error handling so I really don't know and am happy to go with it.

I'd probably have tried the "Error and ErrorKind pair" pattern described in the failure docs to create some sort of `SqlError`.

Reply:
That is what `Error::Sql` is though?